### PR TITLE
Design Picker: Support the free-only filer

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/dotcom-forge/issues/10012

## Proposed Changes

* Introduce the `useIsGoalsHoldout` hook
* Add the "Free only" toggle to the design picker
* Introduce the `useFilteredDesigns` hook to apply the "Free only" toggle to designs
* Introduce the `design-picker/goal-centric` to control the `goal-centric` related features

Note that we will handle the empty result and users with the paid plan in the follow-up PR.

**Screenshots**

image

image

## Why are these changes being made?

* We want to allow users to search free themes

## Testing Instructions

* Assign your account to the `control` variation of the `calypso_onboarding_goals_holdout_20241126` exp. See pbxNRc-4zR-p2.
* Create a new site
* Select any goals
* When you land on the Design Picker, make sure you CAN see the "Free only" toggle
* Enable the "Free only" toggle
* Make sure the paid themes are excluded from the result
* Switch to different categories
* Make sure the toggle still takes effect.
  * TBD: we have to handle the category without any free themes
* Assign your account to the `holdout` variation
* Make sure you CANNOT see the "Free only" toggle

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by fditrapani — APPROVED** [review]
Nice! Very cool. I'm excited to see this in here. I tested it and it works well. Thanks @arthur791004

File: config/horizon.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
 		"current-site/domain-warning": true,
 		"current-site/notice": true,
 		"current-site/stale-cart-notice": false,
+		"design-picker/goal-centric": true,
```
Comment (taipeicoder) [inline config/horizon.json]: Let's keep the feature flag disabled for Horizon for now since we're still early in development.

**Review by taipeicoder — APPROVED** [review]
LGTM 👍 
Left a minor note.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17042785

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @arthur791004 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17042785).

**Pull request merged**